pkg/registry/core/service/storage/rest.go

```diff
@@ -574,7 +574,9 @@ func (r *REST) getAllocatorBySpec(service *api.Service) ipallocator.Interface {
 	}
 
 	secondaryAllocatorCIDR := r.secondaryServiceIPs.CIDR()
-	if *(service.Spec.IPFamily) == api.IPv6Protocol && netutil.IsIPv6CIDR(&secondaryAllocatorCIDR) {
+	secondaryAllocatorIsIPv6 := netutil.IsIPv6CIDR(&secondaryAllocatorCIDR)
+	if *(service.Spec.IPFamily) == api.IPv6Protocol && secondaryAllocatorIsIPv6 ||
```

Comment:
looking at the method 
`func (r *REST) getAllocatorByClusterIP(service *api.Service) ipallocator.Interface {`

https://github.com/kubernetes/kubernetes/pull/89612/files#diff-a17c85ca4491ca68b2c928ed6221828eR556 

I think that it has the same problem and we should fix it there too.